iothub/service/src/Messaging/MessagesClient.cs

```diff
@@ -407,9 +410,28 @@ private void OnConnectionClosed(object sender, EventArgs e)
             }
         }
 
-        private void CheckConnectionIsOpen()
+        private async Task CheckConnectionIsOpenAsync(CancellationToken cancellationToken = default)
         {
-            if (!_amqpConnection.IsOpen)
+            if (_opened)
+            {
+                try
+                {
+                    await _internalRetryHandler
+                        .RunWithRetryAsync(
+                            async () =>
+                            {
+                                await _amqpConnection.OpenAsync(cancellationToken).ConfigureAwait(false);
+                            },
+                            cancellationToken)
+                        .ConfigureAwait(false);
+                    _opened = true;
+                }
+                catch (Exception ex)
+                {
+                    throw new IotHubServiceException(ex.Message, ex);
```

Comment:
Specify the `StatusCode` and `ErrorCode`

Reply:
`throw new IotHubServiceException(ex.Message, HttpStatusCode.RequestTimeout, IotHubServiceErrorCode.RequestTimeout, null, ex);`